test/e2e/operator_test.go

```diff
@@ -1017,6 +1030,11 @@ func TestOperatorUpgrade(t *testing.T) {
 	}
 	t.Logf("observed image %s for deployment %s/contour", main, cfg.SpecNs)
 
+	if err := waitForDaemonsetUpToDate(ctx, kclient, 1*time.Minute, "envoy", cfg.SpecNs, 1); err != nil {
```

Comment:
@sunjayBhatia I don't understand the need for the add'l waits when the Contour is configured for ns removal and the tests currently have a func to wait for the removal of the test ns after the contour is deleted.